Plugins: ES6ify the PluginActivateToggle component

tyxla opened the pull request:
This PR ES6ifies the `PluginActivateToggle` component, converting it into an ES6 class, and fixing all warnings. It also converts the `lib/analytics` usage to the Redux `state/analytics`, and fixes the tests.

To test:

* Checkout this branch
* Go to a page that has a list of plugins (like `http://calypso.localhost:3000/plugins/$site`, where `$site` is one of your Jetpack sites).
* Play around with activation and deactivation of plugins and verify there are no regressions with the display and functionality of the activation toggle.
* Verify the component tests still pass: `npm run test-client client/my-sites/plugins/plugin-activate-toggle/`.

/cc @johnHackworth @ryelle @oskosk

gwwar commented on client/my-sites/plugins/plugin-activate-toggle/index.jsx:
```diff
@@ -1,87 +1,95 @@
+}
+
+PluginActivateToggle.propTypes = {
+	site: PropTypes.object.isRequired,
```
>  Not sure if this is a good idea,

It is 😄 but I can understand if there's too much to untangle at once. Usually, it's a bit easier if we update child components to handle their own data needs first. @tyxla feel free to make a judgement call on if you want to clean up the site passing here or in a follow up PR.

tyxla replied:
Let's do that in another PR. You folks are right, but I have bad memories from doing multipurpose PRs.

obenland commented on client/my-sites/plugins/plugin-activate-toggle/index.jsx:
```diff
@@ -1,87 +1,95 @@
 			} );
 		}
-	},
+	}
```
I couldn't find terribly useful docs on this, but it looks like class properties still need a closing semicolon

gwwar commented on client/my-sites/plugins/plugin-activate-toggle/index.jsx:
```diff
@@ -1,87 +1,95 @@
-				site: this.props.site.ID,
-				plugin: this.props.plugin.slug
+		if ( plugin.active ) {
+			this.props.recordGoogleEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', plugin.slug );
```
We can also create a new variable name if you get tired of typing this out:
```javascript
const { recordGoogleEvent: recordGAEvent } = this.props;
recordGAEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', plugin.slug );
```

johnHackworth commented:
@enejb needs to check this and see that all the cases works right before merging

tyxla commented on client/my-sites/plugins/plugin-activate-toggle/index.jsx:
```diff
@@ -1,87 +1,95 @@
 		] );
 
-		if ( this.props.plugin && 'jetpack' === this.props.plugin.slug ) {
+		if ( plugin && 'jetpack' === plugin.slug ) {
```
I'd vote for keeping this check in case a `null` value gets passed.

obenland replied:
Is that a possibility? 
I'm just wondering what good an activate toggle is if there's no plugin.

tyxla replied:
After doing some tests here and there, I confirm that this is not happening. However, I'm still sticking to my opinion here, because it's possible that such cases might occur while we're migrating plugins to Redux. We've seen that happening with the `getSelectedSite()` migration, for example in #9562.

tyxla commented:
Thanks for the feedback, @gwwar @obenland! I've addressed your comments, and fixed an issue with importing the component within another components. I haven't ES6-ified them, as that would go outside of the PR scope.

Looking forward to some feedback from @enejb as well.

enejb commented:
This works as expected. I tested the toggle in the few different places that it does show up. 
In my testing I found a few different regressions that I will add as issues that should be fixed in a different PR. 

Thanks doing this much needed changes @tyxla. The code is much more readable as a result. 
I think this is ready to :ship:
